packages/vuetify/src/composables/variant.ts

```diff
@@ -0,0 +1,39 @@
+// Composables
+import { useColor } from '@/composables/color'
+
+// Utilities
+import { computed } from 'vue'
+import { propsFactory } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+const allowedVariants = [null, 'default', 'contained', 'outlined', 'text'] as const
+
+type Variant = typeof allowedVariants[number]
+
+export interface VariantProps {
+  color?: String
+  variant?: Variant
+}
+
+// Composables
+export const makeVariantProps = propsFactory({
+  color: String,
+  variant: {
+    type: String as PropType<Variant>,
+    default: 'default',
+    validator: (v: any) => allowedVariants.includes(v),
+  },
+}, 'variant')
+
+export function useVariant (props: VariantProps, name: string) {
+  const variantClasses = computed(() => {
+    return `${name}--variant-${props.variant}`
```

Comment:
```suggestion
    return props.variant ? `${name}--variant-${props.variant}` : ''
```

Reply:
Any component that uses variants should have a default variant class. Same as density